metron-platform/metron-common/src/test/java/org/apache/metron/common/dsl/functions/StringFunctionsTest.java

```diff
@@ -207,4 +207,78 @@ public void testFormatWithNoArguments() throws Exception {
   public void testFormatWithMissingArguments() throws Exception {
     run("FORMAT('missing arg: %d')", Collections.emptyMap());
   }
+
+
+  /**
+  * CHOP StringFunction
+  * @throws Exception
+  */
+  @Test
+  public void testChop() throws Exception {
+    Assert.assertEquals("ab",   run("CHOP('abc')", new HashedMap()));
+    Assert.assertEquals(null,   run("CHOP(null)", new HashedMap()));
+    Assert.assertEquals("abc",  run("CHOP(msg)", ImmutableMap.of("msg", "abc\r\n")));
```

Comment:
Because it's coming in through a variable `msg`.  The idea being that it could be existent in a message flowing through the topologies.  Stellar can't represent `\n` and `\r`, but they can get in there honestly.